prometheus/promhttp/http_test.go

```diff
@@ -331,3 +331,95 @@ func TestHandlerTimeout(t *testing.T) {
 
 	close(c.Block) // To not leak a goroutine.
 }
+
+func BenchmarkEncoding(b *testing.B) {
```

Comment:
Thanks! Do you mind adding correctness tests too (e.g. for weird OfferedCompressions values and zstd?)

Reply:
I've refactored the code into a GetWriter() function to make it easier to test. I still need to add a few testcases and probably get things right with the response writer.